front-app/src/views/Products.tsx

```diff
@@ -0,0 +1,188 @@
+/* eslint-disable react/jsx-key */
+import React, { useEffect, useState } from 'react';
+
+import InfiniteScroll from 'react-infinite-scroller';
+import { useQuery } from 'react-query';
+
+import { getCategories } from '../api/categoriesApi';
+import { getProducts } from '../api/productsApi';
+import SearchIcon from '../assets/svg/search_icon.svg';
+import Vector from '../assets/svg/vector.svg';
+import HelpComponent from '../components/HelpComponent';
+import ProductCard from '../components/ProductCard';
+import TopNav from '../layouts/TopNav';
+
+interface Product {
+  id: number;
+  name: string;
+  price: number;
+  image: string;
+  categoryId: number;
+  createdAt: string;
+  updatedAt: string;
+}
+
+interface Category {
+  id: number;
+  name: string;
+  createdAt: string;
+  updatedAt: string;
+}
+
+const Products = () => {
+  const [queryName, setQueryName] = useState<null | string>(null);
+  const [queryCategories, setQueryCategories] = useState<null | number[]>(null);
+  const [querySortBy, setQuerySortBy] = useState('');
+  const [queryPage, setQueryPage] = useState(0);
+  const [querySize, setQuerySize] = useState(4);
+  const [productList, setProductToList] = useState<Product[]>([]);
+  const [hasMoreValue, setHasMore] = useState(true);
```

Comment:
Same here, this is wrong, this names should always match. `hasMore` and `setHasMore` is good. The word value at the end seems unnecessary to me